test/EndToEndTests/Tests/Client/Microsoft.OData.Client.E2E.Tests/ClientWithoutTypeResolverTests/Tests/OpenTypesWithoutTypeResolverTests.cs

```diff
@@ -0,0 +1,223 @@
+            }
+        }
+
+        public OpenTypesWithoutTypeResolverTests(TestWebApplicationFactory<TestsStartup> fixture)
+            : base(fixture)
+        {
+            _baseUri = new Uri(Client.BaseAddress, "odata/");
+            _context = new Container(_baseUri);
+            _context.HttpClientFactory = HttpClientFactory;
+
+            _context.AddObject("Row", row1);
+            _context.AddObject("Row", row2);
+
+            this.typeNameResolver = _context.ResolveName;
+
+            _context.ResolveName = this.typeNameResolver;
+
+            var rowIndex = new Common.Clients.OpenTypes.RowIndex
+            {
+                Id = TestRowIndexId,
+                Rows = new DataServiceCollection<Common.Clients.OpenTypes.IndexedRow>(_context),
+                DynamicProperties = new Dictionary<string, object>()
+                {
+                    { "IndexComments", "This is a test"}
+                }
+            };
+
+            _context.SaveChanges(SaveChangesOptions.BatchWithSingleChangeset);
+        }
+
+        #region Test Data
+        private Common.Clients.OpenTypes.Row row1 = new Common.Clients.OpenTypes.Row
+        {
+            Id = Guid.NewGuid(),
+            DynamicProperties = new Dictionary<string, object>
+            {
+                { "OpenBoolean", true },
+                { "OpenDateTimeOffset", DateTimeOffset.Now },
+                { "OpenDecimal", decimal.MinusOne },
+                { "OpenDouble", double.NaN },
+                { "OpenFloat", float.PositiveInfinity },
+                { "OpenGuid", Guid.NewGuid() },
+                { "OpenInt16", Int16.MaxValue },
+                { "OpenInt64", Int64.MaxValue },
+                { "OpenString","hello world" },
+                { "OpenTime", TimeSpan.MaxValue }
+            }
+        };
+
+        private Common.Clients.OpenTypes.Row row2 = new Common.Clients.OpenTypes.Row
+        {
+            Id = Guid.NewGuid(),
+            DynamicProperties = new Dictionary<string, object>
+            {
+                { "OpenBoolean", null },
+                { "OpenDateTimeOffset", null },
+                { "OpenDecimal", null },
+                { "OpenDouble", null },
+                { "OpenFloat", null },
+                { "OpenGuid", null },
+                { "OpenInt16", null },
+                { "OpenInt64", null },
+                { "OpenString",null },
+                { "OpenTime", null }
+            }
+        };
+
+        private Common.Clients.OpenTypes.IndexedRow row3 = new Common.Clients.OpenTypes.IndexedRow
+        {
+            Id = Guid.NewGuid(),
+            DynamicProperties = new Dictionary<string, object>()
+            {
+                { "OpenDouble", double.NaN }
+            }
+        };
+        #endregion
+
+        private const int TestRowIndexId = 999;
+        private Func<Type, string> typeNameResolver;
+
+        [Fact, TestPriority(3)]

```

Comment:
I think having tests depend on a specific priority negatively impacts maintainability. If there's no practical way around, I'd suggest to at least re-organize the test methods in order of priority, such that the tests that should be executed first appear first. That way when reading the file from top to bottom, you can follow the logical execution flow of the tests and understand better why/how they depend on each other.

Reply:
I've fixed this